ua-frontend/libs/ua-core/src/+state/ui.actions.ts

```diff
@@ -0,0 +1,11 @@
+export interface LoadData {
```

Comment:
is the `+` with folder names a standard? I have seen that some places and don't particular like it, but if it is a growing standard in the community we can use it. What are the rules for when to use it?

Also, let's talk a bit about state management. We should discuss at a high level and then figure out how the libraries, folders and file structure lays out based on the higher level. Specifically, the way I think about things are something like this:

*Backend*
- There are "resources" that represent backend entities which may or may not be persisted to one or more backend databases.
- A resource has a `schema` (which is a GraphQL data definition) and a TypeScript `model` interface. In most cases the schema and model should be conceptually the same with just syntax differences
- There may or may not be a backend `service` which contains the logic for interacting with a particular resource.

*Frontend*
- Typically the backend data groupings are more granular than the front end. Any given app should have a redux store which contains a set of different types of state (ex auth, metrics, search, etc.)
- An app should be free to define state in whatever way makes sense, but in most cases, it should be a composition of smaller stores pulled together.
- Each store can access one or more backend services
- Each store is accessed via a facade wrapper. We can talk more about this one because I have seen many different implementations of this, but the idea is to cut down on the ngrx cruft by having a layer on top of it which is essentially sugar. So, your components can (in most cases) access data or make changes with one simple line of code. I can show you examples of this.
- I go back and forth on effects. They are very useful in some cases, but also seem like an anti-pattern at times. I need to look at the latest implementation of effects, but in one of my projects I created my own version of this. For this project we should just use effects, but be aware of where you can run into issues.